packages/accounts/src/msca/account/multisigAccount.ts

```diff
@@ -0,0 +1,119 @@
+import type { EntryPointDef } from "@alchemy/aa-core";
+import {
+  createBundlerClient,
+  getAccountAddress,
+  getVersion060EntryPoint,
+  toSmartContractAccount,
+  type Address,
+  type SmartAccountSigner,
+  type SmartContractAccountWithSigner,
+  type UserOperationRequest,
+} from "@alchemy/aa-core";
+import {
+  concatHex,
+  encodeFunctionData,
+  hexToBigInt,
+  type Chain,
+  type Hex,
+  type Transport,
+} from "viem";
+import { MultisigModularAccountFactoryAbi } from "../abis/MultisigModularAccountFactory.js";
+import { multisigMessageSigner } from "../plugins/multisig/signer.js";
+import { getDefaultMultisigModularAccountFactoryAddress } from "../utils.js";
+import { standardExecutor } from "./standardExecutor.js";
+
+export type MultisigModularAccount<
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+> = SmartContractAccountWithSigner<"MultisigModularAccount", TSigner>;
+
+export type CreateMultisigModularAccountParams<
+  TTransport extends Transport = Transport,
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+> = {
+  transport: TTransport;
+  chain: Chain;
+  signer: TSigner;
+  salt?: bigint;
+  factoryAddress?: Address;
+  threshold: bigint;
+  sigs?: Address[];
+  entryPoint?: EntryPointDef<UserOperationRequest>;
+  accountAddress?: Address;
+  initCode?: Hex;
+};
+
+export async function createMultisigModularAccount<
+  TTransport extends Transport = Transport,
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+>(
+  config: CreateMultisigModularAccountParams<TTransport, TSigner>
+): Promise<MultisigModularAccount<TSigner>>;
```

Comment:
nit: can you just put this function's interface on the impl below?

Reply:
this function overloading is required for handling other types of transports below